src/main/electron-types.ts

```diff
@@ -0,0 +1,142 @@
+import * as path from 'node:path';
+
+import { ElectronVersions } from '@electron/fiddle-core';
+import fetch from 'cross-fetch';
+import { IpcMainEvent, app } from 'electron';
+import * as fs from 'fs-extra';
+import packageJson from 'package-json';
+import readdir from 'recursive-readdir';
+import semver from 'semver';
+
+import { ipcMainManager } from './ipc';
+import { NodeTypes } from '../interfaces';
+import { IpcEvents } from '../ipc-events';
+
+let electronTypes: ElectronTypes;
+
+export class ElectronTypes {
+  constructor(
+    private readonly knownVersions: ElectronVersions,
+    private readonly nodeCacheDir: string,
+  ) {}
+
+  public async getNodeTypes(
+    version: string,
+  ): Promise<{ version: string; types: NodeTypes } | undefined> {
+    // Get the Node.js version corresponding to the current Electron version.
+    const v = this.knownVersions.getReleaseInfo(version)?.node;
+
+    if (!v) return;
+
+    try {
+      const dir = this.getCacheDir(v);
+      const ver = await this.cacheAndReturnNodeTypesVersion(v, dir);
+
+      return { version: ver, types: await this.getTypesFromDir(dir) };
+    } catch (err) {
+      console.debug(
+        `Unable to get Node.js types for Electron "${version}": ${err.message}`,
+      );
+      return;
+    }
+  }
+
+  private async getTypesFromDir(dir: string): Promise<NodeTypes> {
+    const types: NodeTypes = {};
+
+    try {
+      const files = (await readdir(dir)).filter((f) => f.endsWith('.d.ts'));
+
+      for (const file of files) {
+        types[path.relative(dir, file) as keyof NodeTypes] = fs.readFileSync(
+          file,
+          'utf8',
+        );
+      }
```

Comment:
nit: no big deal, but this could be async - always a good idea to avoid blocking I/O if we can help it

```suggestion
      for (const file of files) {
        types[path.relative(dir, file) as keyof NodeTypes] = await fs.readFile(
          file,
          'utf8',
        );
      }
```

Reply:
Yep, seems reasonable. When doing these refactors I try to change as little as possible to minimize risk of introducing a new bug, but this is low risk.